:memo: Prettify announcement (#154)
laminlabs/laminr: CHANGELOG.md

# laminr v1.0.0

This is a major rewrite of the package to offer an interface that has feature-parity with the Python API (PR #146).

Please see the updated ["Get started"](https://laminr.lamin.ai/articles/laminr.html) vignette for more information.

Migration guide:

> - Replace `db <- connect()` with `ln <- import_module("lamindb")`, the `ln` object behaves in the same way as the `db` object in laminr < v1

Reviewer: ```suggestion
- Replace `db <- connect()` with `ln <- import_module("lamindb")`. The `ln` object behaves in the same way as the `lamindb` module object in Python and is largely similar to the `db` object in **{laminr}** < v1.
```

Changes to API:

>   - Add an `import_module("lamindb")` function to import **lamindb**

Reviewer: ```suggestion
  - Add an `import_module()` function to import Python modules with additional functionality. Use `import_module("lamindb")` to import **lamindb**.
```

>   - Add an internal `wrap_python()` function to wrap Python objects while replacing Python methods with R methods as needed, leaving most work to `reticulate()`

Reviewer: ```suggestion
  - Add an internal `wrap_python()` function to wrap Python objects while replacing Python methods with R methods as needed, leaving most work to **{reticulate}**
```

  - Update the internal `check_requires()` function to handle Python packages
  - Add custom `cache()`/`load()` methods to the `Artifact` class

>   - Add custom `track()`/`finish()` methods to the **lamindb** module

Reviewer: ```suggestion
  - Add custom `track()`/`finish()` methods to the **lamindb** **module**
  - Add functions for accessing more `lamin` CLI commands
```

# laminr v1.0.0

This is a major rewrite of the package to offer an interface that has feature-parity with the Python API (PR #146).

Please see the updated ["Get started"](https://laminr.lamin.ai/articles/laminr.html) vignette for more information.

> Migration guide:

Author: @lazappi -- I accepted all your suggestions.

To confirm: Do we really only need a single bullet under "Migration guide"?

What does "largely similar" mean? What's the difference?

Author: Rather than doing the re-wording, can you list _what_ changed so that people can update their code and get intuition which calls are differing?

Ideally you make a table

<v1 | >=v1
--- | ---
`hi()` | `hello()`

Author: This is what we did for the lamindb v1 migration:

<img width="882" alt="image" src="https://github.com/user-attachments/assets/214367dd-70a1-4555-955a-2de943af6a0a" />

Reviewer: Added a suggestion with a table of changes from the "Get started" vignette

> 

Reviewer: ```suggestion

- **{laminr}** v1.0.0 requires `lamindb>=1.2`. Run `install_lamindb()` to update the default Python environment.
```

> - Replace `db <- connect()` with `ln <- import_module("lamindb")`. The `ln` object behaves in the same way as the `lamindb` module object in Python and is largely similar to the `db` object in **{laminr}** < v1.

Reviewer: ```suggestion
- Replace `db <- connect()` with `ln <- import_module("lamindb")`. The `ln` object behaves in the same way as the `lamindb` module object in Python and fulfills the same role as the `db` object in **{laminr}** < v1.
- The interface now matches the Python interface. Please refer to the [`lamindb` documentation](https://docs.lamin.ai/) for more information on syntax.
```

Author: Can we say that `db` and `ln` have essentially the same calls rather than pointing the user to the Python documentation?

Is there any case where `db` and `ln` are **not** the same?

Author: 🙏

# laminr v1.0.0

This is a major rewrite of the package to offer an interface that has feature-parity with the Python API (PR #146).

Please see the updated ["Get started"](https://laminr.lamin.ai/articles/laminr.html) vignette for more information.

Migration guide:

- **{laminr}** v1.0.0 requires `lamindb>=1.2`. Run `install_lamindb()` to update the default Python environment.
- Replace `db <- connect()` with `ln <- import_module("lamindb")`. The `ln` object behaves in the same way as the `lamindb` module object in Python and is largely similar to the `db` object in **{laminr}** < v1.

> Changes to API:

Reviewer: ```suggestion
| What | Before | After |
|--------|--------|--------|
| Connect to the default LaminDB instance | `db <- connect()` | `ln <- import_module("lamindb")` |
| Start tracking | `db$track()` | `ln$track()` |
| Get an artifact from another instance | `new_instance <- connect("another/instance"); new_instance$Artifact$get(...)` | `ln$Artifact$using("another/instance")$get(...)` |
| Create an artifact from a path | `db$Artifact$from_path(path)` | `ln$Artifact(path)` | 
| Finish tracking | `db$finish()` | `ln$finish()` |

Changes to API:
```
